Remove local database support (and drop MySQL dependency)

lpetre-ulb opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->

## Description
<!--- Describe your changes in detail -->

This PR completely deletes the no longer used local run database. As consequence the MySQL dependency can be dropped as well as the Python dependency (`gemutils` only for the latter).

### Types of changes
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)

## Motivation and Context
<!--- Why is this change required? What problem does it solve? -->
<!--- If it fixes an open issue, please link to the issue(s) here. -->

Clean up.

## How Has This Been Tested?
<!--- Please describe in detail how you tested your changes. -->
<!--- Include details of your testing environment, and the tests you ran to -->
<!--- see how your change affects other areas of the code, etc. -->

The `cmsgemos` xDAQ applications still compile, run and their basic actions work.

## Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

<!--- Template thanks to https://www.talater.com/open-source-templates/#/page/99 -->

lpetre-ulb committed:
Remove database elements from xDAQ configurations

lpetre-ulb committed:
Delete references to the database in GEMSupervisor

lpetre-ulb committed:
Delete reference to the database in GEMApplication

lpetre-ulb committed:
Delete GEMDatabaseUtils files

lpetre-ulb committed:
Do not link gemutils against MySQL and Python

These libraries were used only for the GEMDatabaseUtils helpers and are
now longer required since its removal.

lpetre-ulb committed:
Remove MySQL dependency

Not needed since the LDQM database removal.

lmoureaux commented:
I still see `-I/usr/include/python2.7` all over the place, is it a build system feature?

lpetre-ulb commented:
> I still see `-I/usr/include/python2.7` all over the place, is it a build system feature?

~~I don't see any in my build tree... Did you start from a clean repo?~~

Oh, in the compilation commands... Good catch! It looks like that every component which used `gemutils` had to add the include directories due to limitations of the build system.

lpetre-ulb committed:
Remove useless Python CFLAGS

Every component which depends on gemutils had to append the Python
headers search paths due to the build system limitations.

lmoureaux commented:
I did `make clean; make` and I see them for `gemonlinedb` (sight), `gemreadout`, `gemhardware/managers`, `gemsupervisor`, `gemdaqmonitor` and obviously `gempython`. I grepped for Python includes and couldn't find anything except in `.d` files. Doesn't block this from being merged.

lpetre-ulb committed:
Delete outdated gemutils DB exceptions

lpetre-ulb commented:
Useless Python includes removed; as well as outdated `gemutils` DB exceptions.

jsturdy commented on gemsupervisor/xml/gemsupervisor_tif_lemo.xml:
```diff
@@ -138,18 +138,6 @@
-          <setupLocation xsi:type="xsd:string">TIF</setupLocation>
-        </DatabaseInfo>
-
 	<rcmsStateListener xsi:type="soapenc:Struct">
```
To be honest, this file (and all others here not named `gemsupervisor.xml` probably should be removed...

lpetre-ulb replied:
Ok, great! I was hesitating to remove them, but I'll remove them as part of my next cleanup PR (targeting at the removal of unused files).

In the future, the configuration files should probably be stored somewhere else (in the `GEMRunConfigs` repository?) or maintained up-to-date the main repository.

jsturdy replied:
Indeed, *actual* configs should be stored where you mention, but the example configs here should be kept to a minimum and stick with the source (and maybe at some point be automatically generated to stay up-to-date with potential code changes)

jsturdy commented on gemsupervisor/xml/gemsupervisor_tif.xml:
```diff
@@ -114,18 +114,6 @@
-          <setupLocation xsi:type="xsd:string">TIF</setupLocation>
-        </DatabaseInfo>
-
 	<rcmsStateListener xsi:type="soapenc:Struct">
```
can remove

jsturdy commented on gemsupervisor/xml/gemsupervisor_ctp7_stress_test.xml:
```diff
@@ -191,18 +191,6 @@
-          <runPeriod xsi:type="xsd:string">2016T</runPeriod>
-          <setupLocation xsi:type="xsd:string">CERN904</setupLocation>
-        </DatabaseInfo>
-
```
can remove

jsturdy reviewed (approved):
I'd still remove them here in this PR, since you're already changing them, but non blocking

lpetre-ulb commented:
Superseded by #324.
